arch/x86/core/cache.c

```diff
@@ -18,32 +18,46 @@
 #include <zephyr/cache.h>
 #include <stdbool.h>
 
+#define X86_REG_CR0_NW BIT(29)
+#define X86_REG_CR0_CD BIT(30)
+
 static inline void z_x86_wbinvd(void)
 {
 	__asm__ volatile("wbinvd;\n\t" : : : "memory");
 }
 
 void arch_dcache_enable(void)
 {
+#ifdef CONFIG_X86_64
+	uint64_t cr0 = 0;
+#else
 	uint32_t cr0;
+#endif
```

Comment:
Can we better use here `unsigned long` ?

Reply:
indeed, tough initialization is only valid for x86_64. I wonder if we have a stricter type for it though (such as uintptr_t )